### PR TITLE
feat: Backlog/unreal/project context/improve opener to follow incoming links

### PR DESCRIPTION
**Pull request by henriknorin-ftrack**
Resolves 

## Changes

- Improved opener to follow incoming links
- Fixed asset list scrolling bug

## Test

Create a new task and link it from an asset build that have openable components, to the task. It should appear in the opener.

### CODE REVIEW

File: source/ftrack_connect_pipeline_qt/ui/utility/widget/asset_version_list_selector.py [hunk 1]
```diff
@@ -146,10 +154,24 @@ def _query_assets_from_context_async(self, context_id, asset_type_name):
...
+            for asset in self.session.query(
+                'select name, versions.task.id, type.id, id, latest_version,'
+                'latest_version.version, latest_version.date '
+                'from Asset where versions.asset.parent.id is {} and type.id is {}'.format(
+                    tcl['from_id'], asset_type_entity['id']
+                )
+            ).all():
+                if not asset['id'] in [_asset['id'] for _asset in assets]:
+                    assets.append(asset)
+
```
Comment (lluisFtrack) [inline source/ftrack_connect_pipeline_qt/ui/utility/widget/asset_version_list_selector.py:174]: All this should happen on a opener plugin or in pipeline, but not in QT. So I should be able to open the correct asset in standalone without the need of having QT.

File: source/ftrack_connect_pipeline_qt/ui/utility/widget/definition_selector.py [hunk 2]
```diff
@@ -284,13 +284,36 @@ def populate_definitions(self):
...
+                                asset_type_name,
+                            )
+                        ).all():
+                            if not asset_version['id'] in [
+                                _asset_version['id']
+                                for _asset_version in asset_versions
+                            ]:
+                                asset_versions.append(asset_version)
+
+                    for asset_version in asset_versions:
```
Comment (lluisFtrack) [inline source/ftrack_connect_pipeline_qt/ui/utility/widget/definition_selector.py:316]: There is to much logic in the QT layer, I should be able to open the correct asset build without using qt, so the logic should,'t be here.

**Review by lluisFtrack — CHANGES_REQUESTED** [review]
We shouldn't have ftrack logic in qt, QT should just be the widgets layer of the framework, nothing else.

File: source/ftrack_connect_pipeline_qt/ui/utility/widget/asset_version_list_selector.py (hunk 1 above)
Comment (henriknorin-ftrack) [reply source/ftrack_connect_pipeline_qt/ui/utility/widget/asset_version_list_selector.py:174]: Will add a TODO so we can revisit this later and move this logic out of QT.

File: source/ftrack_connect_pipeline_qt/ui/utility/widget/definition_selector.py (hunk 2 above)
Comment (henriknorin-ftrack) [reply source/ftrack_connect_pipeline_qt/ui/utility/widget/definition_selector.py:316]: Will add a TODO so we can revisit this later and move this logic out of QT.

**Review by lluisFtrack — APPROVED** [review]

**Pull request merged**